packages/lockfile-lint-api/src/ParseLockfile.js

```diff
@@ -36,11 +36,30 @@ function checkSampleContent (lockfile) {
  */
 function yarnParseAndVerify (lockfileBuffer) {
   const lockfile = yarnParseSyml(lockfileBuffer.toString())
+  const isYarnBerry = typeof lockfile.__metadata === 'object'
   const hasSensibleContent =
-    lockfile && typeof lockfile === 'object' && checkSampleContent(lockfile)
+    lockfile && typeof lockfile === 'object' && checkSampleContent(lockfile, isYarnBerry)
   if (!hasSensibleContent) {
     throw Error('Lockfile does not seem to contain a valid dependency list')
   }
+
+  if (isYarnBerry) {
```

Comment:
realistically this part is where we're really want to branch out to specific functions, each one handling its own lockfile version (yarn1, berry, etc). Again, I don't mind merging this PR and then shooting a new PR to refactor this properly.